pkg/sql/sem/builtins/aggregate_builtins.go

```diff
@@ -209,28 +209,18 @@ var aggregates = map[string]builtinDefinition{
 		func(t *types.T) tree.Overload {
 			info := "Identifies the maximum selected value."
 			vol := tree.VolatilityImmutable
-			// If t is an ambiguous type (like AnyCollatedString), then our aggregate
-			// does not have a fixed return type.
-			if t.IsAmbiguous() {
-				return makeAggOverloadWithReturnType(
-					[]*types.T{t}, tree.FirstNonNullReturnType(), newMaxAggregate, info, vol,
-				)
-			}
-			return makeAggOverload([]*types.T{t}, t, newMaxAggregate, info, vol)
+			return makeAggOverloadWithReturnType(
+				[]*types.T{t}, tree.IdentityReturnType(0), newMaxAggregate, info, vol,
```

Comment:
should this be done for all `collectOverloads` aggregate/window functions?

Reply:
hmm well i think all the other ones are appropriately special cased. avg and sum intentionally have a different return type.

Reply:
i believe i meant items such as percentile_disc_impl, lag, lead, first_value, last_value, nth_value.

Reply:
i see! working on it